git_machete/client.py

```diff
@@ -947,7 +948,7 @@ def print_line_prefix(branch_: LocalBranchShortName, suffix: str) -> None:
             out.write(colored(suffix, edge_color[branch_]))
 
         next_sibling_of_ancestor: List[Optional[LocalBranchShortName]]
-        for branch, next_sibling_of_ancestor in next_sibling_of_ancestor_by_branch:
+        for branch, next_sibling_of_ancestor in next_sibling_of_ancestor_by_branch.items():
```

Comment:
@PawelLipski https://github.com/VirtusLab/git-machete/issues/509 suggests to make `next_sibling_of_ancestor_by_branch` a dict (which i agree with and changed so) and iterate over `self.managed_branches` instead of `next_sibling_of_ancestor_by_branch`. So modified iteration works fine in tests but now `discover` does't work properly locally, i am betting that the order of the branches in `self.managed_branches` is different than in `next_sibling_of_ancestor_by_branch`, but it is still weird that all the tests pass. I will investigate it further

Reply:
Yeah, as i suspected, somehow `self.managed_branches` has different branch order when run with `status` than run with `discover` (which also executes `status()` function after discovering the tree)

Reply:
I guess it's to be expected as `discover` creates tree from scratch and orders branches alphabetically and `status` does dfs. To sum up i think the iteration should stay as it was before, @PawelLipski wdyt?

Reply:
I wasn't aware of the caveat with `discover`... so I'm not insisting, let's stick to iterating over `next_sibling_of_ancestor_by_branch` as it used to be.

Reply:
`dict` -> maybe `OrderedDict` for clarity!